public/js/maho-captcha.js

```diff
@@ -0,0 +1,174 @@
+/**
+ * Maho
+ *
+ * @package     Maho_Captcha
+ * @copyright   Copyright (c) 2025 Maho (https://mahocommerce.com)
+ * @license     https://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ */
+
+const MahoCaptcha = {
+    altchaWidget: null,
+    altchaState: null,
+    frontendSelectors: '',
+    onVerifiedCallback: null,
+
+    async setup(config) {
+        this.altchaWidget = document.querySelector('altcha-widget');
+        this.frontendSelectors = config.frontendSelectors ?? '';
+
+        // Load Altcha JS
+        if (typeof customElements.get('altcha-widget') !== 'function') {
+            await Promise.all([
+                new Promise((resolve, reject) => {
+                    const script = document.createElement('script');
+                    script.src = '/js/altcha.min.js';
+                    script.type = 'module';
+                    script.onload = resolve;
+                    script.onerror = () => reject(`${script.src} Not Found`);
+                    document.head.appendChild(script);
+                }),
+            ]);
+        }
+
+        // Inject Altcha Widget
+        if (!this.altchaWidget) {
+            if (typeof config.widgetAttributes !== 'object' || config.widgetAttributes === null) {
+                throw new Error('widgetAttributes must be specified');
+            }
+            this.altchaWidget = document.createElement('altcha-widget');
+            for (const [ attr, val ] of Object.entries(config.widgetAttributes)) {
+                this.altchaWidget.setAttribute(attr, val);
+            }
+            document.body.appendChild(this.altchaWidget);
+        }
+
+        // Inject Stylesheet
+        if (!document.querySelector('style[maho-captcha-style]')) {
+            const styleEl = document.createElement('style');
+            styleEl.dataset.mahoCaptchaStyle = '';
+            styleEl.textContent = `
+                altcha-widget[name=maho_captcha] {
+                    display: flex;
+                    position: fixed;
+                    bottom: 0;
+                    right: 0;
+                }
+                dialog.maho-captcha-verifying {
+                    position: fixed;
+                    top: 50%;
+                    left: 50%;
+                    transform: translate(-50%, -50%);
+                    display: flex;
+                    flex-direction: column;
+                    align-items: center;
+                    gap: 8px;
+                    padding: 8px;
+                    border: none;
+                    border-radius: 6px;
+                    box-shadow: 0 0 10px rgba(0, 0, 0, 0.3);
+                    body:has(&) {
+                        overflow: hidden;
+                    }
+                    &::backdrop {
+                        background: rgba(0, 0, 0, 0.7);
+                    }
```

Comment:
is this necessary? I didn't really see too much difference and to re-create big js files for new functionality scare me.

Reply:
ok sorry, this is the new feature I couldn't see cause it's not happening to me